dotCMS/src/main/java/com/dotmarketing/portlets/fileassets/business/FileAssetFactoryImpl.java

```diff
@@ -0,0 +1,76 @@
+package com.dotmarketing.portlets.fileassets.business;
+
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionAPI;
+import com.dotmarketing.common.db.DotConnect;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.contentlet.util.ContentletUtil;
+import com.dotmarketing.portlets.folders.model.Folder;
+import com.dotmarketing.util.Logger;
+import com.liferay.portal.model.User;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class FileAssetFactoryImpl implements FileAssetFactory {
+    private final PermissionAPI permissionAPI;
+
+    private final String FIND_FILE_ASSETS_BY_FOLDER_QUERY =
+            "SELECT contentlet.inode " +
+            "FROM identifier, contentlet_version_info, contentlet, structure " +
+            "WHERE identifier.parent_path = ? and " +
+                "identifier.host_inode = ? and " +
+                "contentlet_version_info.identifier = identifier.id and " +
+                "contentlet_version_info.%s_inode = contentlet.inode and " +
+                "contentlet.structure_inode = structure.inode and " +
+                "structure.structuretype =4;";
+
+    public FileAssetFactoryImpl() {
+        permissionAPI = APILocator.getPermissionAPI();
+    }
+
+    @Override
+    public List<Contentlet> findFileAssetsByFolderInDB(
+            final Folder parentFolder,
+            final  User user,
+            final boolean live) throws DotDataException, DotSecurityException {
+
+        final boolean respectFrontendRoles = live;
+        final DotConnect dotConnect = new DotConnect();
+        dotConnect.setSQL(String.format(FIND_FILE_ASSETS_BY_FOLDER_QUERY, live ? "live" : "working"));
+        dotConnect.addParam(parentFolder.getPath());
+        dotConnect.addParam(parentFolder.getHostId());
+
+        if(!permissionAPI.doesUserHavePermission(parentFolder, PermissionAPI.PERMISSION_READ, user, respectFrontendRoles)){
```

Comment:
this should be the first line, well before the dotConnect

Reply:
done https://github.com/dotCMS/core/pull/18225/commits/b3221ca6fdad2fc09e713c11a6b62f4339571aac#diff-159930580f03a629fdc4eb9506576d39R50